### DISCUSSION TITLE
[Bug]: Not able to quickstart stimzi

### DISCUSSION THREAD

**Discussion by Devesh-N** [strimzi/strimzi-kafka-operator#9022] [General]
### Bug Description

Hello all, I am a Kubernetes noob, and I wanted to deploy Kafka on Strimzi. In order to do that I followed this URL on the Strimzi website
[32.0 quicstart docs](https://strimzi.io/docs/operators/0.32.0/quickstart#proc-product-downloads-str). I got everything set up but was stuck on this step **"Wait for the cluster to be deployed:**

**kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n my-kafka-project"** part of the documentation. Just so you know, I have 5 nodes in my AWS EKS Kubernetes cluster, those nodes being t2.micro each. I have also enabled OIDC IAM, as this solution was mentioned in stack overflow [error solution](https://stackoverflow.com/questions/68725070/k8s-pods-failure-error-while-running-volumebinding-prebind-plugin-for-pod-a#:~:text=returned%20nothing.-,Enable%20OIDC%20IAM,-Enabled%20OIDC%20IAM). The error I get is 
`Events:`
  `Type     Reason            Age                From               Message`
 ` ----     ------            ----               ----               -------`
  `Warning  FailedScheduling  14s (x3 over 10m)  default-scheduler  running PreBind plugin "VolumeBinding": binding volumes:` `provisioning failed for PVC "data-my-cluster-zookeeper-0"` 
 or something else. I have followed everything in the documentation, step by step, command by command. What do I do?
 Additional info
 K8s version 1.27
Strimzi version 0.32.0

### Steps to reproduce

Follow the Quickstart docs link

### Expected behavior

Stuff should run

### Strimzi version

0.32.0

### Kubernetes version

1.27

### Installation method

as stated in docs

### Infrastructure

Amazon EKS

### Additional context

Please help me ASAP

**Comment by sreejesh123** [comment]
Looks like issue with PVC binding .  Is the storage attached to your cluster working and accessible for the service account ? I had a similar (not same ) issue where my pvc was not provisioned due to storage restriction . I would check the volume
Claim and see if their is any erro on k8s cluster level ..

**Comment by Devesh-N** [comment]
> Looks like issue with PVC binding . Is the storage attached to your cluster working and accessible for the service account ? I had a similar (not same ) issue where my pvc was not provisioned due to storage restriction . I would check the volume Claim and see if their is any erro on k8s cluster level ..

@sreejesh123 How can I check? each one is gp3 volume along with 80 gb storage

**Comment by scholzj** [comment]
@Devesh-N I do not think there is any need to ask the same question on at least 4 different places. Please pick one and close the others.

**Comment by Devesh-N** [comment]
> @Devesh-N I do not think there is any need to ask the same question on at least 4 different places. Please pick one and close the others.

I don't know where to ask, I am new to all of this

**Comment by Devesh-N** [comment]
And can you please help me @scholzj

**Comment by Devesh-N** [comment]
@scholzj Do you require any additional information?

**Comment by scholzj** [comment]
I think @sreejesh123 already pointed you in the right direction. It looks like the storage provisioning failed. But it is hard to say what exactly the problem is without access to your Kubernetes cluster. Strimzi only creates the PVC resources and the rest is up to Kubernetes.

**Comment by scholzj** [comment]
Triaged on the community call on 24th August: This does not look like a bug but like an environment issue. Should be converted to discussion.

**Comment by Devesh-N** [comment]
Hello @scholzj Thanks for the reply. How am I supposed to use strimzi on EKS? And docs in particular?

**Reply by scholzj** [reply]
From what you provided, everything points out to the storage not being provisioned. That is something your Kubernetes cluster does. Strimzi just creates the Persistent Volume Claims - your Kubernetes creates the persistent volumes and binds / mounts them.

**Comment by Devesh-N** [comment]
and how can I join the community meetings? @scholzj

**Reply by scholzj** [reply]
You can check out the details in https://github.com/strimzi/strimzi-kafka-operator#strimzi-community-meetings

**Comment by hiroarabay** [comment]
@Devesh-N @scholzj 
I got same the issue and same versions (k8s, strimzi).

**Reply by hiroarabay** [reply]
sorry I am using strimzi-0.36.1. same kubernetes version 1.27.

**Reply by Devesh-N** [reply]
> sorry I am using strimzi-0.36.1. same kubernetes version 1.27.

Okay let me know when you get solution

**Reply by Devesh-N** [reply]
@hiroarabay did you find anything yet?

**Reply by hiroarabay** [reply]
@Devesh-N No sorry

**Reply by Devesh-N** [reply]
> @Devesh-N No sorry 

@hiroarabay I had solved it a while back, but forgot about it. I remember watching a stackoverflow answer telling me to install the AWS csi or some sort of volume addon

**Reply by scholzj** [reply]
You need AWS EBS CSI driver to provision the storage. But I would expect it to be available out of the box is you use Amazon EKS, or?

**Reply by Devesh-N** [reply]
> You need AWS EBS CSI driver to provision the storage. But I would expect it to be available out of the box is you use Amazon EKS, or?

@scholzj no it isn't available. The addon needs to be added manually. Even after it is added you need to give the addon something sort of permission

**Reply by scholzj** [reply]
Hmm, it is a long time since I used EKS. I wonder why they do it -> shouldn't they make it as easy as possible to consume more AWS services and pay them more money? :-o

**Reply by Devesh-N** [reply]
> Hmm, it is a long time since I used EKS. I wonder why they do it -> shouldn't they make it as easy as possible to consume more AWS services and pay them more money? :-o

@scholzj Maybe. AWS does want money. They do make it easy to consume more AWS services but that is just not the case for EKS

**Reply by hiroarabay** [reply]
@scholzj @Devesh-N 
Hi I've resolved my issue.

Solution
1. Create the aws EBS csi manually.
https://docs.aws.amazon.com/eks/latest/userguide/csi-iam-role.html
2. Managing the Amazon EBS CSI driver as an Amazon EKS add-on
https://docs.aws.amazon.com/eks/latest/userguide/managing-ebs-csi.html
3. restart ebs-csi-controller 
```
kubectl delete pods -n kube-system -l=app=ebs-csi-controller
```
https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/1024

**Reply by Devesh-N** [reply]
Bro I swear I solved it 4 hours after your reply and just saw it after 8 hours @hiroarabay 😭

**Reply by Devesh-N** [reply]
@hiroarabay But still my cluster is not accessible outside the nodes. Any fix for that?

**Reply by hiroarabay** [reply]
@Devesh-N 
you need to edit `aws-auth` and add your iam role in the config.
or 
your need to recreate the EKS with your aws account. then you can access your eks cluster. (your iam role become default eks role)
You don't have a EKS permission.
https://docs.aws.amazon.com/ja_jp/eks/latest/userguide/add-user-role.html

**Reply by Devesh-N** [reply]
Okay @hiroarabay I will check it out. You seem like a cool person, should we connect?

**Comment by Devesh-N** [comment]
Alright. issue resolved
